contracts/strategies/liquity/LQTYStakingStrategy.sol

```diff
@@ -0,0 +1,312 @@
+     */
+    function emergencyExit() public onlyGovernance {
+        _setPausedInvesting(true);
+        exitRewardPool();
+    }
+
+    /**
+     * @dev Resumes the ability to invest into the underlying rewards pool.
+     */
+    function continueInvesting() public onlyGovernance {
+        _setPausedInvesting(false);
+    }
+
+    /**
+     * @dev Can completely disable liquidation of DODO rewards. This can be
+     *      useful in case of emergency exits.
+     */
+    function setSell(bool _sell) public onlyGovernance {
+        _setSell(_sell);
+    }
+
+    /**
+     * @dev Set the minimum amount of DODO needed to trigger a sale (for a liquidation).
+     */
+    function setSellFloor(uint256 _sellFloor) public onlyGovernance {
+        setSellFloor(_sellFloor);
+    }
+
+    function finalizeUpgrade() external onlyGovernance {
+        _finalizeUpgrade();
+
+        // Reset the liquidation paths - they need to be reset manually
+        storedLiquidationPaths[usdc][weth] = new address[](0);
+        storedLiquidationDexes[usdc][weth] = new bytes32[](0);
+        storedLiquidationPaths[weth][lqty] = new address[](0);
+        storedLiquidationDexes[weth][lqty] = new bytes32[](0);
+    }
+
+    // ---------------- Internal methods ----------------
+
+    function rewardPoolBalance() internal view returns (uint256 balance) {
+        balance = ILQTYStaking(rewardPool()).stakes(address(this));
+    }
+
+    function claimRewards() internal {
+        if (rewardPoolBalance() > 0) {
+            ILQTYStaking(rewardPool()).unstake(0);
+        }
+    }
+
+    function enterRewardPool() internal {
+        uint256 currentBalance = IERC20(underlying()).balanceOf(address(this));
+        ILQTYStaking(rewardPool()).stake(currentBalance);
+    }
+
+    function exitRewardPool() internal {
+        uint256 staked = rewardPoolBalance();
+        if (staked > 0) {
+            ILQTYStaking(rewardPool()).unstake(staked);
+        }
+    }
+
+    /**
+     * @dev Stakes everything the strategy holds into the reward pool.
+     */
+    function investAllUnderlying() internal onlyNotPausedInvesting {
+        // This check is needed for avoiding reverts in case we stake 0
+        if (IERC20(underlying()).balanceOf(address(this)) > 0) {
+            enterRewardPool();
+        }
+    }
+
+    function _liquidateReward() internal {
+        // Profits can be disabled for possible simplified and rapid exit
+        if (!sell()) {
+            emit ProfitsNotCollected(sell(), true);
+            return;
+        }
+
+        uint256 lusdBalance = IERC20(lusd).balanceOf(address(this));
+        if (lusdBalance > 0) {
+            // First, liquidate the LUSD rewards to USDC on Curve
+            // as that's where most of the LUSD liquidity resides
+            IERC20(lusd).safeApprove(lusd3CrvPair, 0);
+            IERC20(lusd).safeApprove(lusd3CrvPair, lusdBalance);
+            ICurveMetapool(lusd3CrvPair).exchange_underlying(
```

Comment:
I might be wrong but Curve doesn't seem to be supported by the Universal Liquidator yet. If it were, this swap could be made via the UL instead.